src/hotspot/share/gc/shenandoah/shenandoahHeap.hpp

```diff
@@ -689,6 +684,7 @@ class ShenandoahHeap : public CollectedHeap {
   HeapWord* allocate_memory_under_lock(ShenandoahAllocRequest& request, bool& in_new_region);
   HeapWord* allocate_from_gclab_slow(Thread* thread, size_t size);
   HeapWord* allocate_new_gclab(size_t min_size, size_t word_size, size_t* actual_size);
+  bool retry_allocation(size_t original_full_gc_count) const;
```

Comment:
It feels like this should be called `should_retry_allocation()`. Also a 1-line documentation comment: ``` // We want to retry an unsuccessful attempt at allocation until at least a full gc. ```

Reply:
PS: do we know that a full gc will reclaim all soft refs? I suppose that is Shenandoah policy?

Reply:
Yes, I like that name better. And yes, full gc will reclaim all soft refs.